### PR TITLE
chore: drop support for GKE 1.25 and AKS 1.25, 1.26

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
### Checklist

- [X] Changelog updated or skip changelog label added
- [X] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -94,8 +94,8 @@ The following table displays the tested Kubernetes and Helm versions.
 | K8s with EKS           | 1.25<br/>1.26<br/>1.27<br/>1.28<br/>1.29 |
 | K8s with EKS (fargate) | 1.25<br/>1.26<br/>1.27<br/>1.28<br/>1.29 |
 | K8s with Kops          | 1.25<br/>1.26<br/>1.27<br/>1.28<br/>1.29 |
-| K8s with GKE           | 1.25<br/>1.26<br/>1.27<br/>1.28<br/>1.29 |
```
Comment (sumo-drosiek) [inline docs/README.md:97]: Why do we drop gke 1.25? I see e2e tests passing for it

File: docs/README.md [hunk 2]
```diff
@@ -94,8 +94,8 @@ The following table displays the tested Kubernetes and Helm versions.
...
-| K8s with AKS           | 1.25<br/>1.26<br/>1.27<br/>1.28<br/>1.29 |
```
[the remark this answers is not in the document]
Comment (aboguszewski-sumo) [reply docs/README.md:98]: Then we should drop it first probably? Docs say that it's no longer supported: https://learn.microsoft.com/en-us/azure/aks/supported-kubernetes-versions?tabs=azure-cli#aks-kubernetes-release-calendar

File: docs/README.md (hunk 1 above)
Comment (aboguszewski-sumo) [reply docs/README.md:97]: It has reached EOL, we should drop the tests then: https://cloud.google.com/kubernetes-engine/docs/release-schedule#schedule-for-release-channels

**Review by sumo-drosiek — APPROVED** [review]
Let's proceed

**Pull request merged**